### PR TITLE
[Test] Build WithKyuubiServerOnKuberntes

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Refactor KyuubiOnKubernetesTestsSuite
Build WithKyuubiServerOnKuberntes like WithKyuubiServerOnYarn

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+package org.apache.kyuubi.kubernetes.test
+
+import io.fabric8.kubernetes.client.DefaultKubernetesClient
+
+import org.apache.kyuubi.WithKyuubiServer
+import org.apache.kyuubi.config.KyuubiConf
+
+trait WithKyuubiServerOnKubernetes extends WithKyuubiServer {
+  protected val kyuubiServerConf: KyuubiConf
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:26]: it should not be a required method

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala [hunk 2]
```diff
@@ -31,34 +31,11 @@ import org.apache.kyuubi.operation.SparkQueryTests
...
+  override protected val kyuubiServerConf: KyuubiConf = KyuubiConf()
+  override protected val connectionConf: Map[String, String] = Map(
+    "spark.master" -> "local",
+    "spark.executor.instances" -> "1")
 
-  override protected def jdbcUrl: String = {
-    assert(_jdbcUrl != null, "Failed to get Kyuubi server")
-    _jdbcUrl
-  }
+  override protected def jdbcUrl: String = getJdbcUrl
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala:40]: let's override it in parquet class

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/ProcessUtils.scala [hunk 3]
```diff
@@ -14,6 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/ProcessUtils.scala:17]: revert

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala (hunk 2 above)
Comment (zwangsheng) [reply integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala:40]: This method required by `JDBCTestHelper`, pass by `SparkQueryTests`, should be here.

Comment (ulysses-you) [reply integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala:38]: ah I see

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          .withName(kyuubiServer.getMetadata.getName)
+          .getLog
+      throw new IllegalStateException(
+        s"Kyuubi server pod state error: ${kyuubiServer.getStatus.getPhase}, log:\n$log")
+    }
+    val kyuubiServerIp = MiniKube.getIp
+    val kyuubiServerPort =
+      kyuubiServer.getSpec.getContainers.get(0).getPorts.get(0).getHostPort
+    val _jdbcUrl = s"jdbc:hive2://$kyuubiServerIp:$kyuubiServerPort/;"
+    assert(_jdbcUrl != null, "Failed to get Kyuubi server")
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:61]: this assert is unnecessary that it must be non null

Comment (ulysses-you) [reply integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/WithKyuubiServerOnKubernetes.scala:61]: just return `s"jdbc:hive2://$kyuubiServerIp:$kyuubiServerPort/;"`

File: integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala [hunk 5]
```diff
@@ -31,34 +30,10 @@ import org.apache.kyuubi.operation.SparkQueryTests
...
-          .getLog
-      throw new IllegalStateException(
-        s"Kyuubi server pod state error: ${kyuubiServer.getStatus.getPhase}, log:\n$log")
-    }
-    val kyuubiServerIp = MiniKube.getIp
-    val kyuubiServerPort =
-      kyuubiServer.getSpec.getContainers.get(0).getPorts.get(0).getHostPort
-    s"jdbc:hive2://$kyuubiServerIp:$kyuubiServerPort/;"
-  }
+class KyuubiOnKubernetesTestsSuite extends WithKyuubiServerOnKubernetes with SparkQueryTests {
```
Comment (ulysses-you) [inline integration-tests/kyuubi-kubernetes-it/src/test/scala/org/apache/kyuubi/kubernetes/test/deployment/KyuubiOnKubernetesTestsSuite.scala:33]: KyuubiOnKubernetesWithLocalSpark

**Comment by zwangsheng** [comment]
Seems CI test fail, have nothing to do with kubernetes - it test.

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**